test/testdata/resolver/recover_from_bad_superclass.rb

```diff
@@ -0,0 +1,37 @@
+# typed: true
+
+def make_a_class
+  Class.new
+end
+
+class A < 1 # error: Superclasses must only contain constant literals
+end
+T.let(A, Class)
+A.new.class
+
+DynamicParent = make_a_class
+class B < DynamicParent # error: Superclasses and mixins may only use class aliases like `A = Integer`
+end
+T.let(B, Class)
+B.new.class
+
+Alias = T.type_alias(Object)
+class C < Alias # error: Superclasses and mixins may not be type aliases
+end
+T.let(C, Class)
+C.new.class
+
+class D < D # error: Circular dependency: `D` is a parent of itself
+end
+T.let(D, Class)
+D.new.class
+
+class F < G
+end
+T.let(F, Class)
+F.new.class
+
+class G < F # error: Circular dependency: `G` and `F` are declared as parents of each other
+end
+T.let(G, Class)
+G.new.class
```

Comment:
Is there a reason why we don't have any new tests with modules? It seems like we'd want to also test the StubMixin code path.

Reply:
Ah maybe I'm misunderstanding. Was it the case that we used to **only** recover from bad mixins, and not bad superclasses?

Reply:
We were recovering for both in the same way -- by stubbing the bad thing with a module. This made it so that we had classes whose superclass is a module in the cases here. These degenerate classes failed the `T.let(Degenerate, Class)` assertion and didn't have `.new` methods.

We already have tests for recovering from bad mixins like `circular_mixin`.